packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js

```diff
@@ -403,4 +442,110 @@ if (!ShadowRoot.prototype.createElement) {
+    const attachInternals = HTMLElement.prototype['attachInternals'];
+    const methods = [
+      'setFormValue',
+      'setValidity',
+      'checkValidity',
+      'reportValidity',
+    ];
+
+    HTMLElement.prototype['attachInternals'] = function (...args) {
+      const internals = attachInternals.call(this, ...args);
+      internalsToHostMap.set(internals, this);
+      return internals;
+    };
+
+    methods.forEach((method) => {
+      const proto = window['ElementInternals'].prototype;
+      const originalMethod = proto[method];
+
+      proto[method] = function (...args) {
+        const host = internalsToHostMap.get(this);
+        const definition = definitionForElement.get(host);
+        if (definition['formAssociated'] === true) {
+          originalMethod?.call(this, ...args);
+        } else {
+          throw new DOMException(
+            `Failed to execute ${originalMethod} on 'ElementInternals': The target element is not a form-associated custom element.`
+          );
+        }
+      };
+    });
+
+    // Emulate the native RadioNodeList object
+    class RadioNodeList extends Array {
+      constructor(elements) {
+        super(...elements);
+        this._elements = elements;
+      }
+
+      get value() {
+        return this._elements.find((element) => element.checked.value);
+      }
+    }
+
+    // Emulate the native HTMLFormControlsCollection object
+    class HTMLFormControlsCollection {
+      constructor(elements) {
+        const entries = new Map();
+        elements.forEach((element, index) => {
+          const name = element.getAttribute('name');
+          this[+index] = element;
+          if (name && entries.has[name]) {
+            entries.get(name).push(element);
+          } else {
+            entries.set(name, [element]);
+          }
+        });
+        this['length'] = elements.length;
+        entries.forEach((value, key) => {
+          if (!value) return;
+          if (value.length === 1) {
+            this[key] = value[0];
+          } else {
+            this[key] = new RadioNodeList(value);
+          }
+        });
+      }
+
+      ['namedItem'](key) {
+        return this[key];
+      }
+    }
+
+    // Override the built-in HTMLFormElements.prototype.elements getter
+    const formElementsDescriptor = Object.getOwnPropertyDescriptor(
+      HTMLFormElement.prototype,
+      'elements'
+    );
+
+    Object.defineProperty(HTMLFormElement.prototype, 'elements', {
+      get: function () {
+        const nativeElements = formElementsDescriptor.get.call(this, []);
+
+        const include = [];
+
+        for (const element of nativeElements) {
+          const definition = definitionForElement.get(element);
+
+          // Only purposefully formAssociated elements or built-ins will feature in elements
+          if (
+            (definition && definition['formAssociated'] === true) ||
+            element.localName.indexOf('0') === -1
```

Comment:
But, I guess as long as it doesn't have a `definition`, it's ok to let an element filter through, probably don't need the tag name check at all. So maybe just
```suggestion
            !definition || definition['formAssociated'] === true
```

Reply:
Done!